syntax = "proto3";
package enricher.v1;

import "google/protobuf/timestamp.proto"; // <1>

message City { // <2>
  // Universal Unique Identifier of the city.
  string uuid = 1;
  // City name.
  string name = 2;
  // The country name where the city is located.
  string country = 3;
  // The continent name where the country is located.
  string region = 4;
}

message WeatherForecast {
  // The city this weather forecast is for.
  City city = 1; // <3>
  // Seconds of UTC time since Unix epoch 1970-01-01T00:00:00Z.
  google.protobuf.Timestamp timestamp = 2; // <4>
  // Amount of rain forecasted for today.
  double rain_sum_mm = 3;
  // Amount of showers forecasted for today.
  double showers_sum_mm = 4;
  // Amount of snow forecasted for today.
  double snowfall_sum_cm = 5;
  // Maximum temperature forecasted for today.
  double temperature_max_celsius = 6;
  // Minimum temperature forecasted for today.
  double temperature_min_celsius = 7;
}

message EnrichRequest {
  // The weather forecast details for a city.
  WeatherForecast weather_forecast = 1;
}

message EnrichResponse {
  // Text content of the enriched weather forecast.
  string content = 1;
}

// The service enriches request information.
service EnricherService {
  // This method returns the enriched content of the request.
  rpc Enrich(EnrichRequest) returns (EnrichResponse);
}
